frameworks/hdfs/tests/test_sanity.py

```diff
@@ -135,9 +135,13 @@ def test_kill_data_node():
 @pytest.mark.sanity
 @pytest.mark.recovery
 def test_kill_scheduler():
-    sdk_cmd.kill_task_with_pattern('hdfs.scheduler.Main', shakedown.get_service_ips('marathon').pop())
-    config.check_healthy(service_name=sdk_utils.get_foldered_name(config.SERVICE_NAME))
+    foldered_name = sdk_utils.get_foldered_name(config.SERVICE_NAME)
+    task_id_service_name = sdk_utils.get_task_id_service_name(foldered_name)
+    old_scheduler_id = sdk_tasks.get_task_ids(foldered_name, task_id_service_name)[0]
 
+    sdk_cmd.kill_task_with_pattern('hdfs.scheduler.Main', shakedown.get_service_ips('marathon').pop())
+    config.expect_recovery(service_name=foldered_name)
```

Comment:
Killing the scheduler shouldn't trigger a recovery as recovery implies that something is being done on the nodes themselves.

When killing the scheduler, we would expect:
* The scheduler task is restarted
* The other tasks are **not** restarted
* There is no in-progress deployment or recovery